### ISSUE TITLE
Failed to add layer for python 3.11

### ISSUE THREAD

**Issue by vhnguyenae** [DataDog/datadog-lambda-python#364]
## Expected Behavior
Datadog layer is generated for the lambda function

## Actual Behavior
Datadog layer is not gemerated for the lambda function

## Steps to Reproduce the Problem

  1. Using [aws-lambda-python-alpha v 2.63.0-alpha.0](https://www.npmjs.com/package/@aws-cdk/aws-lambda-python-alpha)
  2. Changing [runtime](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-lambda-python-alpha/lib/function.ts#L22) from 3.10 to 3.11
  3. Running `cdk diff`

## Specifications

  - Datadog Lambda Layer version: 77
  - Python version: 3.11
  - datadog-cdk-constructs: [v2-1.7.4](https://github.com/DataDog/datadog-cdk-constructs/releases/tag/v2-1.7.4)
  
## Stacktrace
  
  ```
[~] AWS::CloudFormation::Stack LambdaStack.NestedStack/LambdaStack.NestedStackResource LambdaStackNestedStackLambdaStackNestedStackResource574FF764 
 ├─ [~] NestedTemplate
 │   └─ [~] .Resources:
 │       ├─ [~] .CheckConnectionFunctionB51290BD:
 │       │   ├─ [~] .Metadata:
 │       │   │   └─ [~] .aws:asset:path:
 │       │   │       ├─ [-] asset.cd90685942b11db240383848503ec75b28f2c004b72c04850a1c53fa761b6711
 │       │   │       └─ [+] asset.2e42a50c5b67e58c8443e57b289fd9f14c8963607f34aea90939a47be5963718
 │       │   └─ [~] .Properties:
 │       │       ├─ [~] .Code:
 │       │       │   └─ [~] .S3Key:
 │       │       │       ├─ [-] cd90685942b11db240383848503ec75b28f2c004b72c04850a1c53fa761b6711.zip
 │       │       │       └─ [+] 2e42a50c5b67e58c8443e57b289fd9f14c8963607f34aea90939a47be5963718.zip
 │       │       ├─ [~] .Environment:
 │       │       │   └─ [~] .Variables:
 │       │       │       └─ [~] .DD_TAGS:
 │       │       │           ├─ [-] name:ga4-checkConnectionFunction,git.commit.sha:ac2bcf082cfd7a7176fdba778efc85db4645b956,git.repository_url:github.com/onpage-org/ga4.git
 │       │       │           └─ [+] name:ga4-checkConnectionFunction,git.commit.sha:ca043e8e96e7ddc23a96e9c4d9d84a712cd6dc68,git.repository_url:github.com/onpage-org/ga4.git
 │       │       ├─ [~] .Handler:
 │       │       │   ├─ [-] datadog_lambda.handler.handler
 │       │       │   └─ [+] check_connection.handler
 │       │       ├─ [-] Removed: .Layers
 │       │       └─ [~] .Runtime:
 │       │           ├─ [-] python3.10
 │       │           └─ [+] python3.11

  ```

**Comment by vhnguyenae** [comment]
Hi @astuyve , thank you, I bump both `aws-lambda-python-alpha` to `2.93` and `datadog-cdk-constructs` to `2-1.8.0`.

The deployment went successfully, I checked the runtime configuration of my Lambda Function and it was 3.11 as expected.
<img width="880" alt="image" src="https://github.com/DataDog/datadog-lambda-python/assets/60604412/57057eca-5bc2-4dd4-b8a4-93b15c921caf">

However, checking the Datadog log, I found some error related to 3.10 below:
Not sure if we missed anything?
```
[ERROR] HTTPError: HTTP Error 404: Not Found
Traceback (most recent call last):
  File "/opt/python/lib/python3.10/site-packages/datadog_lambda/wrapper.py", line 214, in __call__
    self.response = self.func(event, context, **kwargs)
  File "/var/task/check_connection.py", line 4, in handler
```
